src/main/java/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package service;
+
+import dao.ShopDao;
+import java.io.File;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private ShopDao shopDao;
+    private TransactionService transactionService;
+
+    public ShopServiceImpl(ShopDao shopDao, TransactionService transactionService) {
+        this.shopDao = shopDao;
+        this.transactionService = transactionService;
+    }
+
+    @Override
+    public void process(File fromFile, File toFile) {
+        List<String> data = shopDao.readFromFile(fromFile);
+        transactionService.processData(data);
+        String report = transactionService.generateReport();
+        shopDao.writeToFile(toFile, report);
+    }
+}
```

Comment:
You don't need this service either. You have to create methods according to the solid principle. They shouldn't call each other in a chain instead you should create services with independent logic and call them in the main method of the Main class you didn't create